cmake/compiler_flags.cmake

```diff
@@ -0,0 +1,5 @@
+if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+  # workaround for MacOS 10.13 and below which does not support std::visit
+  # see https://github.com/OpenMS/OpenMS/issues/5714
+  add_definitions(-D_LIBCPP_DISABLE_AVAILABILITY)
+endif()
```

Comment:
@dbaston I'm not really sure where to add this. Tried there but It didn't work

Reply:
just pushed a commit putting it in the root CMakeLists